src/zenml/model/model_version.py

```diff
@@ -346,6 +346,73 @@ def metadata(self) -> Dict[str, "MetadataType"]:
             for name, response in response.run_metadata.items()
         }
 
+    def delete_artifact(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        only_link: bool = True,
+        delete_metadata: bool = True,
+        delete_from_artifact_store: bool = False,
+    ) -> None:
+        """Delete the artifact linked to this model version.
+
+        Args:
+            name: The name of the artifact to delete.
+            version: The version of the artifact to delete (None for latest/non-versioned)
+            only_link: Whether to only delete the link to the artifact.
+            delete_metadata: Whether to delete the metadata of the artifact.
+            delete_from_artifact_store: Whether to delete the artifact from the artifact store.
+        """
+        from zenml.client import Client
+        from zenml.models import ArtifactVersionResponse
+
+        artifact_version = self.get_artifact(name, version)
+        if isinstance(artifact_version, ArtifactVersionResponse):
+            client = Client()
+            client.delete_model_version_artifact_link(
+                model_version_id=self.id,
+                artifact_version_id=artifact_version.id,
+            )
+            if not only_link:
+                client.delete_artifact_version(
+                    name_id_or_prefix=artifact_version.id,
+                    delete_metadata=delete_metadata,
+                    delete_from_artifact_store=delete_from_artifact_store,
+                )
+
+    def delete_all_artifacts(
+        self,
+        only_link: bool = True,
+        delete_metadata: bool = True,
+        delete_from_artifact_store: bool = False,
+    ) -> None:
+        """Delete all artifacts linked to this model version.
+
+        Args:
+            only_link: Whether to only delete the link to the artifact.
+            delete_metadata: Whether to delete the metadata of the artifact.
+            delete_from_artifact_store: Whether to delete the artifact from the artifact store.
+        """
+        from zenml.client import Client
+
+        client = Client()
+
+        if not only_link:
+            mv = self._get_model_version()
+            ids = mv.data_artifact_ids
+            ids.update(mv.model_artifact_ids)
+            ids.update(mv.deployment_artifact_ids)
+
+        client.delete_all_model_version_artifact_links(self.id)
+
+        if not only_link:
+            for id_ in ids:
+                client.delete_artifact_version(
+                    name_id_or_prefix=id_,
+                    delete_metadata=delete_metadata,
+                    delete_from_artifact_store=delete_from_artifact_store,
+                )
+
```

Comment:
Is deleting model artifacts (in addition to links) ever going to be supported through the dashboard ? If yes, then I'd suggest doing this directly in the sql zen store (i.e. passing the "only_link" attribute through the REST API as a query param). There is also the added value of only making one API call instead of calling the API multiple times for multiple artifacts (which can be a performance issue).

This looks good otherwise.

Reply:
https://zenml.atlassian.net/browse/OSSK-54 and https://zenml.atlassian.net/browse/OSSK-153 are in our backlog

Reply:
@strickvl bare in ind that OSSK-153 is being tracked in more detail here: https://zenml.atlassian.net/jira/software/projects/PROD/boards/11/backlog?epics=visible&text=delete 

@stefannica in general the answer is yes we would like to allow deleting model versions and links of artifacts to model versions via the dashboard

Reply:
@stefannica , thanks for the suggestion, I looked a bit closer into this and cannot fully agree that it will fit nicely in the SQL Sstore because we also have the option to delete the actual artifact (`delete_from_artifact_store` flag), which deals with Stack Component directly and belongs to the client IMO, so I made a move to use Client if actual deletion of artifacts is requested, otherwise it will be processed on SQL Engine in packed queries (one query with other table lookup didn't work due to some SQLAlchemy nonsense, so kept it in 2).

Reply:
@avishniakov makes total sense, thank you !